### PR TITLE
release v0.16.0

### PR DESCRIPTION
**Pull request by acatangiu**
Updated the version number, changelog, cargo dependencies & credits.md.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "firecracker"
-version = "0.15.0"
+version = "0.16.0"
 authors = ["Amazon firecracker team <firecracker-devel@amazon.com>"]
```
Comment (raduweiss) [inline Cargo.toml]:
```suggestion
authors = ["Amazon Firecracker team <firecracker-devel@amazon.com>"]
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,13 +1,15 @@
 # Changelog
 
-## [Unreleased]
+## [0.16.0]
 
 ### Added
 
+- Added [experimental] AMD support.
```
Comment (raduweiss) [inline CHANGELOG.md]: Do we mention what's different / experimental about AMD anywhere? Off the top of my head, it's experimental because we haven't done exhaustive benchmarking / testing, and the main difference are the extra 200 ms Linux guest start-up time (due to the lapic difference).

Comment (acatangiu) [reply CHANGELOG.md]: @serban300 Would you please add a few lines in `README.md` describing the AMD status-quo?

Comment (serban300) [reply CHANGELOG.md]: I'm not sure `README.md` is the best place to add this description. I can't see it fitting well there. Should we add it in `CHANGELOG.md` ? Or somewhere else ?

Anyway Radu's description is accurate.

Comment (dianpopa) [reply CHANGELOG.md]: We can at least update the Prerequisites section of the [Getting Started](https://github.com/firecracker-microvm/firecracker/blob/master/docs/getting-started.md). We are mentioning that we only support x86 hosts.

File: CHANGELOG.md [hunk 3]
```diff
@@ -27,6 +31,7 @@
 ### Fixed
 
 - Corrected the conditional compilation of the seccomp rule for `madvise`.
+- Corrected the seccomp filter when building with glibc.
```
Comment (dianpopa) [inline CHANGELOG.md]: I think this is not worth mentioning. I would mention it only if correcting this would result in succesful glibc compilation. If that is the case right now, I would rephrase this to focus on that instead.

File: CHANGELOG.md (hunk 2 above)
Comment (serban300) [reply CHANGELOG.md]: AMD is still x86. You mean we should add something about ARM support ?

Comment (andreeaflorescu) [reply CHANGELOG.md]: @raduweiss I believe we have no AMD tests in the CI. So testing it once doesn't really count as testing from my perspective. Why do we actually want to say we have experimental AMD support? What is the benefit here? With no testing on an AMD platform the support can literally be broken by any commit.

File: CHANGELOG.md (hunk 3 above)
Comment (andreeaflorescu) [reply CHANGELOG.md]: The commit that was adding the seccomp whitelist for glibc was tested so that Firecracker built with glibc successfully runs a VM. It's not about just building Firecracker.

File: CHANGELOG.md (hunk 2 above)
Comment (raduweiss) [reply CHANGELOG.md]: In the general case, I think we will learn a lot faster when we have more potential sources of feedback. Having the `[experimental]` tag on AMD makes it possible to get input from outside the maintainer team. This is important because the reverse unknown is true as well: we can test all we want on a specific CI machine, only to find out that we didn't consider sets of relevant use cases. As an example, I'll point to the (lack of) huge pages support that was discovered right after we open sourced.

As for support breaking with each commit, I'll argue that for this particular case the chances are low, the added code is either a/ refactoring in the code paths that get exercised in Intel as well,  and so are actually tested right now b/ code specific to AMD that won't be broken by other development. Also, there are some tests, e.g., for [cpu features](https://github.com/firecracker-microvm/firecracker/blob/master/tests/integration_tests/functional/test_cpu_features.py).

We definitely can't call this un-experimental until all the items in the milestone are done.

Comment (andreeaflorescu) [reply CHANGELOG.md]: @raduweiss  The tests exist, but they're not run because we don't spin up any AMD instances. Essentially whatever is inside the `if cpu_vendor == CpuVendor.AMD:` block is never executed in our CI hence we aren't testing any AMD functionality. Furthermore, we don't test that any platform independent functionality of Firecracker works on AMD.

I think we should learn from the vsock experience and not call things experimental without adding at least minimal testing (e.g. one microvm boots). At the very least I would specify what experimental means for us so people know what to expect (as you also mentioned in your first comment).

Comment (raduweiss) [reply CHANGELOG.md]: Well, in my mind experimental means "if you're a customer that's interested in this feature, and are willing to work with a very early release, you can now start trying things out in advance." Not sure if that's what others read into that 😄.

I still think this is really valuable for the reasons mentioned in the post above. As another example, we got tons of early feedback from Lambda last spring/summer, and that helped us build a better Firecracker. That feedback would have arrived quite late in the cycle if we waited and only shared a well tested build.

So, should we just talk a bit about what `[experimental]` means, to set the right expectations? Or use more traditional `[alpha]`/`[beta]` tags?

Comment (acatangiu) [reply CHANGELOG.md]: I've marked the AMD support as `[alpha]`. I think that is self-explanatory in regards to benchmarking and testing.

@raduweiss regarding the longer boot-time on AMD, that is not something we should have in our documents, we have an open issue on it in GitHub and that's where we track it.

@firecracker-microvm/compute-capsule please re-review.

File: CHANGELOG.md (hunk 3 above)
Comment (acatangiu) [reply CHANGELOG.md]: Yes, now Firecracker correctly works when built with glibc (although since we don't have functional tests for glibc, there is no guarantee on how long this will be the case).
Since this was something broken and now works correctly, I think it's worth including in the changelog.

Comment (andreeaflorescu) [reply CHANGELOG.md]: One more thing: wasn't the documentation added in the same PR? I noticed that this entry is in 0.15.2 while the documentation for glibc is in 0.16.0.
